auto_gptq/nn_modules/qlinear/qlinear_qbits.py

```diff
@@ -0,0 +1,320 @@
+import math
+
+import numpy as np
+import torch
+import torch.nn as nn
+import transformers
+from ...utils.import_utils import QBITS_AVAILABLE
+from logging import getLogger
+
+
+logger = getLogger(__name__)
+
+if QBITS_AVAILABLE:
+    from intel_extension_for_transformers import qbits  # with QBits kernels ()
+
+BITS_DTYPE_MAPPING = {
+    4: "int4_clip",
+    8: "int8",
+}
+
+
+def convert_dtype_torch2str(dtype):
+    if dtype == torch.int8:
+        return "int8"
+    elif dtype == torch.float:
+        return "fp32"
+    elif dtype == torch.float16:
+        return "fp16"
+    elif dtype == torch.bfloat16:
+        return "bf16"
+    elif isinstance(dtype, str) and dtype in ["int8", "fp32", "fp16", "bf16"]:
+        return dtype
+    else:
+        assert False, "Unsupported pytorch dtype {} to str dtype".format(dtype)
+
+
+class QuantLinear(nn.Module):
+    QUANT_TYPE = "qbits"
+
+    def __init__(
+        self,
+        bits,
+        group_size,
+        infeatures,
+        outfeatures,
+        bias,
+        kernel_switch_threshold=128,
+        trainable=False,
+        weight_dtype=torch.bfloat16,
+        **kwargs,
+    ):
+        super().__init__()
+
+        if bits not in [4, 8]:
+            raise NotImplementedError("Only 4,8 bits are supported for QBits.")
```

Comment:
we support 2bit linear now and it works pretty well in auto-round.
looks like we don't need to modify too much qbits_qlinear code besides adding a new BITS_DTYPE_MAPPING